components/ai_chat/resources/untrusted_conversation_frame/components/assistant_response/web_sources_event.tsx

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import Icon from '@brave/leo/react/icon'
+import createSanitizedImageUrl from '$web-common/create_sanitized_image_url'
+import formatMessage from '$web-common/formatMessage'
+import { getLocale } from '$web-common/locale'
+import * as mojom  from '../../../common/mojom'
+import { useUntrustedConversationContext } from '../../untrusted_conversation_context'
+import styles from './web_sources_event.module.scss'
+
+const UNEXPANDED_SOURCES_COUNT = 4;
+
+function WebSource (props: { source: mojom.WebSource }) {
+  const context = useUntrustedConversationContext()
+
+  const { source } = props
+
+  const handleOpenSource = (e: React.MouseEvent, source: mojom.WebSource) => {
+    e.preventDefault()
+    context.uiHandler?.openURLFromResponse(source.url)
+  }
+
+  const host = new URL(source.url.url).hostname
+  return (
+    <li>
+      <a href={source.url.url} title={source.title} onClick={(e) => handleOpenSource(e, source)}>
+        <img src={createSanitizedImageUrl(source.faviconUrl.url)} />
```

Comment:
Am I understanding right that we are loading the favicon as an `chrome-untrusted://image?src=...` here?
Do we have a proxied database of favicons we can get from `search`?

Reply:
I'm not sure I understand what you're asking @thypon. We are using the Search API which is returning sources used in the search. Each source has a favicon url. We are passing that favicon url to the image sanitizer here.

Reply:
rephrasing: are these favicon the ones proxied by Brave Search itself, or are the original favicon url of the website?
In the second case we are introducing a leak through favicon download.

Reply:
The API is only returning proxied or pre-cached urls from the search API directly. The browser is only allowing imgs.search.brave.com items.